src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -717,14 +717,21 @@ private static int CalculateNumberOfExceptions(__ExceptionInfo[] excp)
             stackSize = m_stackSize;
             if (m_exceptionHeader != null && m_exceptionHeader.Length != 0)
             {
-                if (m_exceptionHeader.Length < 4)
+                // If length < 4, fail. Below pattern elides future bounds checks in method.
+                if (3 >= (uint)m_exceptionHeader.Length)
                     throw new FormatException();
 
                 byte header = m_exceptionHeader[0];
 
                 if ((header & 0x40) != 0) // Fat
                 {
-                    EHCount = (BinaryPrimitives.ReadInt32LittleEndian(m_exceptionHeader.AsSpan(1)) - 4) / 24;
+                    Span<byte> size = stackalloc byte[4];
+                    size[0] = m_exceptionHeader[1]; // n.b. this is a 24-bit integer, not a 32-bit integer
```

Comment:
Is it actually an improvement to copy this over and then read it again? Reading the 3 bytes and shifting them to the right place would seem better.